### PR TITLE
Data modelling

### PR DESCRIPTION
**Pull request by ArBond**

### CODE REVIEW

File: Lab08.MVC/Domain/JourneyRoute.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    public class JourneyRoute
+    {
+        public long JourneyRouteId { get; set; }
+
+        public int Length { get; set; }
+        public string StartLocation { get; set; }
+        public string FinishLocation { get; set; }
+        public string SourceRouteSchemaPath { get; set; }
+
+        public virtual ICollection<Journey> Journeys { get; set; }
```
Comment (Glareone) [inline Lab08.MVC/Domain/JourneyRoute.cs]: no reason to set this binding.

Journey 1 - N Journey routes?

if yes:

journey:

        public virtual ICollection<JourneyRoute> JourneyRoutes { get; set; } 

and that's it.

if one-one\ N-one:

 public virtual JourneyRoute JourneyRoute { get; set; }

File: Lab08.MVC/Domain/Password.cs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+﻿namespace Lab08.MVC.Domain
+    public class Password
+    {
+        public long PasswordId { get; set; }
+
+        public byte[] Salt { get; set; }
+        public byte[] PasswordHash { get; set; }
+
+        public User User { get; set; }
```
Comment (Glareone) [inline Lab08.MVC/Domain/Password.cs]: delete this relation.

user has a password, not the opposite

File: Lab08.MVC/Domain/User.cs [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+{
+    public class User
+    {
+        public long UserId { get; set; }
+
+        public string Login { get; set; }
+        public string FirstName { get; set; }
+        public string LastName { get; set; }
+        public string NickName { get; set; }
+        [RegularExpression("^([0-9]+)$", ErrorMessage = "Invalid Phone Number")]
```
Comment (Glareone) [inline Lab08.MVC/Domain/User.cs]: ^([0-9\+\\-\\(\\)]+)$

File: Lab08.MVC/Domain/User.cs [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+        public string Phone { get; set; }
+        public DateTime CreationTime { get; set; } = DateTime.Now;
+
+        public virtual UserRating Rating { get; set; }
+        public virtual Password Password { get; set; }
+        public virtual ICollection<Journey> OrganizedJourneys { get; set; }
```
Comment (Glareone) [inline Lab08.MVC/Domain/User.cs]: but you have:

public virtual User Organizer { get; set; }

do something with this)

File: Lab08.MVC/Domain/User.cs (hunk 3 above)
Comment (ArBond) [reply Lab08.MVC/Domain/User.cs]: ;)

File: Lab08.MVC/Domain/User.cs (hunk 4 above)
Comment (ArBond) [reply Lab08.MVC/Domain/User.cs]: canceled

File: Lab08.MVC/Domain/Password.cs (hunk 2 above)
Comment (ArBond) [reply Lab08.MVC/Domain/Password.cs]: done

File: Lab08.MVC/Domain/JourneyRoute.cs (hunk 1 above)
Comment (ArBond) [reply Lab08.MVC/Domain/JourneyRoute.cs]: done

**Pull request merged**